docs/docs/migration-guide.mdx

```diff
@@ -24,6 +26,916 @@ If you still have training data in Markdown format then the recommended approach
+  <TabItem value="old">
+
+  ```python
+  class MyPolicy(Policy):
+      ...
+  ```
+
+  </TabItem>
+  <TabItem value="new">
+
+  ```python {3-5}
+  from rasa.engine.recipes.default_recipe import DefaultV1Recipe
+
+  @DefaultV1Recipe.register(
+      DefaultV1Recipe.ComponentType.POLICY_WITH_END_TO_END_SUPPORT,
+      is_trainable=True
+  )
+  class MyPolicy(Policy):
+      ...
+  ```
+
+  </TabItem>
+</Tabs>
+
+**Providing Rule-only Data to a Policy**
+
+Rasa Open Source allows excluding [forms](forms.mdx) or [slots](domain.mdx#slots) which
+are completely handled by
+[rules](rules.mdx) from becoming features in other policies.
+In Rasa Open Source 2 this information was passed onto the
+policies using the `set_shared_policy_states` method which set the policy attribute
+`_rule_only_data`. Rasa Open Source passes the names of rule-only slots and forms via the
+`predict_action_probabilities` method. The passed `rule_only_data` can be `None`
+in case the [`RulePolicy`](policies.mdx#rule-policy) is not part of your model
+configuration.
+
+<Tabs values={[{"label": "Rasa 2.0 (old)", "value": "old"}, {"label": "Rasa 3.0 (new)", "value": "new"}]}  groupId="3-0-policy-rule-only-data" defaultValue="new">
+  <TabItem value="old">
+
+  ```python
+  class MyPolicy(Policy):
+
+      def set_shared_policy_states(self, **kwargs: Any) -> None:
+          """Sets policy's shared states for correct featurization."""
+          self._rule_only_data = kwargs.get("rule_only_data", {})
+  ```
+
+  </TabItem>
+  <TabItem value="new">
+
+  ```python {6}
+  class MyPolicy(Policy):
+      def predict_action_probabilities(
+          self,
+          tracker: DialogueStateTracker,
+          domain: Domain,
+          rule_only_data: Optional[Dict[Text, Any]] = None,
+    ) -> PolicyPrediction:
+  ```
+
+  </TabItem>
+</Tabs>
+
+#### Changes to Custom NLU Components
+
+**Inheriting from `GraphComponent`**
+
+NLU components which previously inherited from one of the following classes additionally
+need to inherit from the
+[`GraphComponent` interface](custom-graph-components.mdx#the-graphcomponent-interface):
+- `SparseFeaturizer`
+- `DenseFeaturizer`
+- `IntentClassifier`
+- `EntityExtractor`
+- `Component`
+
+This snippet shows the required changes:
+
+<Tabs values={[{"label": "Rasa 2.0 (old)", "value": "old"}, {"label": "Rasa 3.0 (new)", "value": "new"}]}  groupId="3-0-nlu-parent-class" defaultValue="new">
+  <TabItem value="old">
+
+  ```python
+  class MyNLUComponent(SparseFeaturizer):
+      ...
+  ```
+
+  </TabItem>
+  <TabItem value="new">
+
+  ```python {3}
+  from rasa.engine.graph import GraphComponent
+
+  class MyNLUComponent(GraphComponent, SparseFeaturizer):
+      ...
+  ```
+
+  </TabItem>
+</Tabs>
+
+** Inheriting from `EntityExtractorMixin` instead of `EntityExtractor`**
+
+The `EntityExtractor` class was renamed to `EntityExtractorMixin`:
```

Comment:
This made me ponder. Why is `EntityExtractorMixin` a thing while we don't have `IntentClassifierMixin`?

Reply:
If there's a reason for this inconsistency, it'd be good to briefly mention it.

Reply:
In my opinion makes sense to rename `EntityExtractorMixin` for consistency but should be done in a separate PR (cc @joejuzl )